### ISSUE TITLE
Bug when searching for git package

### ISSUE THREAD

**Issue by TatriX** [racer-rust/racer#804]
I have a file `main.rs`
```rust
extern crate ggez;
...
```
and in `Cargo.toml`
```
ggez = { git = "https://github.com/ggez/ggez.git" }
```
I call `racer find-definition 1 13 src/main.rs`
and get:
```
ERROR:racer::cargo: ERROR!: Error { repr: Os { code: 2, message: "No such file or directory" } } /home/tatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/racer-2.0.10/src/racer/cargo.rs 511
ERROR:racer::cargo: ERROR!: Error { repr: Os { code: 2, message: "No such file or directory" } } /home/tatrix/.cargo/registry/src/github.com-1ecc6299db9ec823/racer-2.0.10/src/racer/cargo.rs 511
END
```
I found that `racer` tries to load the file
```
"/home/tatrix/.cargo/git/checkouts/ggez-goodies-3c34bc3ae60a9788/master/.git/refs/heads/master"
```
which is wrong.
So I deleted that directory, because I don't need `ggez-goodies` crate, and now `racer` works fine. But it's a bug anyway.

**Comment by genrtul** [comment]
I had a similar issue with a crate in a git repo.
in Cargo.toml I had:

mycrate = { git = "ssh://myhost/mycrate.git" }

The crate that depended on mycrate compiled correctly, however I couldn't complete any symbols defined in mycrate. Completion for crates from crates.io worked fine.
Racer was giving this output on stderr when I tried completing symbols belonging to mycrate:

racer::cargo: ERROR!: Os { code: 2, kind: NotFound, message: "No such file or directory" } src/racer/cargo.rs 492

I deleted all cached copies of mycrate in ~/.cargo/git/checkouts and completion started working again.
